sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/TsConfigProvider.java

```
@@ -177,26 +176,61 @@ public List<String> tsconfigs(SensorContext context) {
+        return singletonList(tsconfigFile);
+      } catch (IOException e) {
+        LOG.warn("Generating wildcard tsconfig failed", e);
+        return emptyList();
+      }
+    }
+
+    static String writeFile(String content) throws IOException {
+      var tempFile = Files.createTempFile(null, null);
+      Files.writeString(tempFile, content, StandardCharsets.UTF_8);
+      return tempFile.toAbsolutePath().toString();
+    }
+
+    private static Map<String, List<String>> defaultTsConfig = new ConcurrentHashMap<>();
+
     private final TempFolder folder;
     private final Function<FileSystem, FilePredicate> filePredicateProvider;
     private final Map<String, Object> compilerOptions;
+    private final FileWriter fileWriter;
 
     DefaultTsConfigProvider(TempFolder folder, Function<FileSystem, FilePredicate> filePredicate) {
       this(folder, filePredicate, new HashMap<>());
     }
 
     DefaultTsConfigProvider(TempFolder folder, Function<FileSystem, FilePredicate> filePredicate, Map<String, Object> compilerOptions) {
+      this(folder, filePredicate, compilerOptions, DefaultTsConfigProvider::writeFile);
+    }
+
+    DefaultTsConfigProvider(TempFolder folder, Function<FileSystem, FilePredicate> filePredicate, Map<String, Object> compilerOptions, FileWriter fileWriter) {
       this.folder = folder;
       this.filePredicateProvider = filePredicate;
       this.compilerOptions = compilerOptions;
+      this.fileWriter = fileWriter;
     }
 
     @Override
     public List<String> tsconfigs(SensorContext context) throws IOException {
       if (context.runtime().getProduct() == SonarProduct.SONARLINT) {
-        // we don't support per analysis temporary files in SonarLint see https://jira.sonarsource.com/browse/SLCORE-235
-        LOG.warn("Generating temporary tsconfig is not supported in SonarLint context.");
-        return emptyList();
+        return createDefaultTsConfig(context, compilerOptions, fileWriter);
```

I think we can just inline `createDefaultTsConfig` because it's a bit misleading since whatever comes after the if statement also creates a default TSConfig file.

---

Inlined